### ISSUE TITLE
Mask password in Error output while executing http request

### ISSUE THREAD

**Issue by prathish2503** [prometheus/prom2json#124]
When we send url := "http://user:pass@<prometheus url to get metrics>" to FetchMetricFamilies function, If there is an error say connection timeout the error in is printed along with username and password:

**Comment by prathish2503** [comment]
Hi @beorn7 ,
Could you please advice if this is possible...
Thank you.

**Comment by beorn7** [comment]
I'm not sure if masking the output log is worth it for something that you have just sent over the network in an unencrypted fashion as part of the URL.

As you said, it is much better to use something like `req.SetBasicAuth`. We could extend `FetchMetricFamilies` to accept a request or something. But I'm afraid that more and more very specific requirements will show up. `FetchMetricFamilies` is actually not doing a lot before calling `ParseResponse`.

Would it work for you to recreate what `FetchMetricFamilies` is doing in your own code, including setting the auth method of choice, and then call `ParseResponse` yourself?

**Comment by prathish2503** [comment]
Thanks for taking a look @beorn7 

I just thought it would be useful for most users to be able to provide credentials if the url endpoint needs basic auth.

Would give a try to recreate FetchMetricFamilies in my local code with auth (thought that it would be just kinda duplicate code) :)

**Comment by beorn7** [comment]
Yeah, but it's really not a lot of duplicated code. There are certainly more details that users might want to tweak, and creating more and more complicated versions of `FetchMetricFamilies` might be more effort than just duplicating a few lines of code.

(Note that we cannot just change the signature of `FetchMetricFamilies` as this would break existing users.)

**Comment by beorn7** [comment]
With no movement here in awhile, I assume the current state is good enough and close this.
